app/src/main/java/com/zulip/android/networking/response/events/GetEventResponse.java

```diff
@@ -0,0 +1,81 @@
+package com.zulip.android.networking.response.events;
+
+import android.support.annotation.Nullable;
+
+import com.google.gson.annotations.SerializedName;
+import com.zulip.android.util.TypeSwapper;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by patrykpoborca on 8/26/16.
+ */
+
+public class GetEventResponse {
+
+    @SerializedName("msg")
+    private String msg;
+
+    @SerializedName("result")
+    private String result;
+
+    @SerializedName("handler_id")
+    private int handlerId;
+
+    @SerializedName("events")
+    private List<EventsBranch> events;
+
+    @Nullable
+    public <T>List<T> getEventsOf(EventsBranch.BranchType branchType) {
+        return getEventsOf(branchType, null);
+    }
+
+    @Nullable
+    public <T extends EventsBranch, R>List<R> getEventsOf(EventsBranch.BranchType branchType, TypeSwapper<T, R> converter) {
+        if(events == null) {
+            return null;
+        }
+        try {
+            List<R> types = new ArrayList<>(events.size());
+            for (int i = 0; i < events.size(); i++) {
+                if (events.getClass().equals(branchType.getKlazz())) {
+                    EventsBranch orig = events.get(i);
+                    if(converter != null) {
+                        types.add(converter.convert((T) orig));
+                    }
+                    else {
+                        types.add((R) orig);
+                    }
+                }
+            }
+            return types;
+        }
+        catch(Exception e) {
+            //catch misuse
```

Comment:
Not sure, I think it's fine to have things reported back to the server. But part of the reason I didn't use `ZLog` too much was due to me not knowing if that was your preference or not. 

I think it's actually okay to make some noise in crashlytics with small non-fatal exceptions so that we know what's not behaving according to design.

Thoughts?